pkg/containerutil/containerutil.go

```diff
@@ -371,6 +372,13 @@ func Stop(ctx context.Context, container containerd.Container, timeout *time.Dur
 
 	task, err := container.Task(ctx, cio.Load)
 	if err != nil {
+		// NOTE: NotFound doesn't mean that container hasn't started.
+		// In docker/CRI-containerd plugin, the task will be deleted
+		// when it exits. So, the status will be "created" for this
+		// case.
+		if errdefs.IsNotFound(err) {
```

Comment:
Note that we possibly have similar problems in other places in our codebase, where we assume that a container does not exist because we fail retrieving the task.